### PR TITLE
Recursevely go over all submodules

### PR DESCRIPTION
**Pull request by matanhavi**
Using git submodules command going over all the submodules and not only the ones on the whitelist
and fixed white spaces
Fixes issue #6271

### CODE REVIEW

**Comment by dagar** [comment]
Looks good. Have you tried this on OSX? The builds run with GIT_SUBMODULES_ARE_EVIL, so I just wanted to make sure this works with the ancient version of bash in OSX.

**Comment by matanhavi** [comment]
No, I didn't check it on OSX

**Comment by dagar** [comment]
On OSX

     % make
    Tools/check_submodules.sh: line 76: mapfile: command not found

     % sh --version
    GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)
    Copyright (C) 2007 Free Software Foundation, Inc.

File: Tools/check_submodules.sh [hunk 1]
```diff
@@ -61,34 +59,26 @@ then
...
-	check_git_submodule src/lib/DriverFramework
-	check_git_submodule src/lib/DriverFramework/cmake/cmake_hexagon
-	check_git_submodule src/lib/DriverFramework/dspal
-	check_git_submodule src/lib/ecl
-	check_git_submodule src/lib/matrix
-	check_git_submodule src/modules/uavcan/libuavcan
-	check_git_submodule unittests/googletest
-	check_git_submodule src/drivers/gps/devices
+	declare -a submodules=$(git submodule status --recursive | awk '{ print $2 }')
+	mapfile -t arr <<< "$submodules"
```
Comment (dagar) [inline Tools/check_submodules.sh]: mapfile not available on OSX

Comment (matanhavi) [inline Tools/check_submodules.sh]: from one link i found that:
"Since bash4, this can be done even more efficiently with the mapfile builtin"

**Comment by dagar** [comment]
Thanks!

**Pull request merged**